src/qamqpchannel.cpp

```diff
@@ -210,14 +219,19 @@ void QAmqpChannelPrivate::closeOk(const QAmqpMethodFrame &)
     Q_Q(QAmqpChannel);
     Q_EMIT q->closed();
     q->channelClosed();
-    opened = false;
+    newState(ChannelClosedState);
 }
 
 void QAmqpChannelPrivate::openOk(const QAmqpMethodFrame &)
 {
     Q_Q(QAmqpChannel);
     qAmqpDebug(">> OpenOK");
-    opened = true;
+
+    newState(ChannelOpenState);
+    if (qosDefined)
+        q->qos(requestedPrefetchCount,
```

Comment:
http://wiki.qt.io/Qt_Coding_Style#Line_breaks, we have 100 columns to work with so I think this fits on a single line